### ISSUE TITLE
Document recovery from full disk

### ISSUE THREAD

**Issue by varac** [borgbackup/borg#1262]
I filled my backup partition with borg and now even borg list is crashing.
The documentation should descript how to recover from a full disk. 
[This comment](https://github.com/borgbackup/borg/issues/1114#issuecomment-222656759) suggests:

```
you can try deleting the segment files of the last incomplete backup and then run borg check --repair. After that, use borg delete or prune with the --save-space flag.
```

But i don't know how to safely do this and am afraid of destroying my whole backup.

**Comment by enkore** [comment]
Try this script

```py
import sys

from borg.repository import Repository

with Repository(sys.argv[1], lock=False) as repository:
    lio = repository.io
    for segment, segment_file in lio.segment_iterator():
        if lio.is_committed_segment(segment):
            print('Committed', segment_file)
```

It doesn't lock so doesn't write and should work on a full disk. It should give you a list of "committed" segments. If your repository has segments beyond the last committed segment you can rollback manually by deleting these **up to but not including** the last committed segment.

(If you have the master branch installed, make it `is_committed_segment(segment_file)` instead)

**Comment by varac** [comment]
hugh, doesn't work for me:

```
varac@mail:~$ python3 ~/borg-free-space.py /var/backup/borg/
Traceback (most recent call last):
  File "/home/varac/borg-free-space.py", line 5, in <module>
    with Repository(sys.argv[1], lock=False) as repository:
AttributeError: __exit__

```

maybe it's because i have an old borg version installed (borg 0.29.1.dev23+ngc5dcf46)?

**Comment by enkore** [comment]
Yes, that Borg version is too old for that script, it needs 1.0.1 or newer.

**Comment by ThomasWaldmann** [comment]
@enkore "up to" could be misunderstood.

**Comment by enkore** [comment]
This should be solved with #1263 and #1272 / #1336. If there are still corner cases that are currently not handled a new report should be created.
